ckan/lib/files/base.py

```diff
@@ -0,0 +1,267 @@
+from __future__ import annotations
+
+import dataclasses
+from collections.abc import Iterable
+from typing import Any
+
+import file_keeper as fk
+from typing_extensions import TypeAlias
+
+from ckan.config.declaration import Declaration
+from ckan.config.declaration.key import Key
+
+Capability: TypeAlias = fk.Capability
+Location: TypeAlias = fk.Location
+
+FileData: TypeAlias = fk.FileData
+MultipartData: TypeAlias = fk.MultipartData
+
+
+def is_supported_type(content_type: str, supported: Iterable[str]) -> bool:
+    """Check whether content_type matches supported types.
+
+    :param content_type: tested type
+    :param supported: collection of supported types
+    """
+    maintype, subtype = content_type.split("/")
```

Comment:
Is it worth adding some error handling or checks on `content_type` here (to ensure it follows  "type/format") or this has been handled before by file-keeper?